### DISCUSSION TITLE
Integrating a Piecewise Function

### DISCUSSION THREAD

**Discussion by abrombo** [sympy/sympy#23956] [Q&A]
I am integrating a piecewise (polynomials in general), f(t), with a moving gate of length one, that is setting the limits of integration to [t0-1/2,t0+1/2].  The result is then the convolution of the gate function with f(t).  I know the answer is a piecewise polynomial of one degree higher than the original f(t) but the answer from integrate is not in the form of a piecewise function.  Is there anything I can do to convert the output of integrate to a piecewise function?

![convolve](https://user-images.githubusercontent.com/58143428/185951511-daffe7e3-7d5a-4e46-97a2-de2cac7056f1.jpg)

**Comment by oscarbenjamin** [comment]
What version of SymPy are you using?

I don't think I see the same as you (using master):
```python
In [1]: y = symbols('y', real=True)

In [2]: g = Piecewise((0, y <= -0.5), (1, y <= 0.5), (0, y >= 0.5))

In [3]: g
Out[3]: 
⎧0  for y ≤ -0.5
⎪               
⎨1  for y ≤ 0.5 
⎪               
⎩0  for y ≥ 0.5 

In [4]: integrate(g, y)
Out[4]: 
⎧   0     for y ≤ -0.5
⎪                     
⎨y + 0.5  for y ≤ 0.5 
⎪                     
⎩  1.0     otherwise  
```
I might have missed something in your code though. (It's better to post code in a form ready for copy paste rather than a screenshot.)
